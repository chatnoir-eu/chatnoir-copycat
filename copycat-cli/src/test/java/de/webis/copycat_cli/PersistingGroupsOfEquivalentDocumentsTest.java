package de.webis.copycat_cli;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;

import org.approvaltests.Approvals;
import org.junit.Test;

import lombok.SneakyThrows;
import net.sourceforge.argparse4j.inf.Namespace;

public class PersistingGroupsOfEquivalentDocumentsTest {

	@Test
	public void approveSmallSampleWithMultipleTopicsAndSmallGroups() {
		Approvals.verify(runOnFile("small-sample-multiple-topics-only-small-groups.jsonl"));
	}
	
	@Test
	public void approveSmallSampleWithMultipleTopicsAndLargeGroups() {
		Approvals.verify(runOnFile("multiple-topics-larger-groups.jsonl"));
	}
	
	@Test
	public void approveToucheExampleTopicTask1() {
		Approvals.verify(runOnFile("touche21-topic91.jsonl"));
	}
	
	@SneakyThrows
	String runOnFile(String inputFile) {
		Path outputFile = Files.createTempFile("unit-test-copycat", ".jsonl");
		Namespace args = argsForInputFile("src/test/resources/test-pairwise-near-duplicates/" +inputFile, outputFile.toFile());
		App.persistGroupsOfEquivalentDocuments(args);
		String ret = new String(Files.readAllBytes(outputFile));
		outputFile.toFile().delete();
		
		return ret;
	}
	
	private static Namespace argsForInputFile(String inputFile, File outputFile) {
		return CliArguments.parseArgs(new String[] {
			"--documents", "AnseriniIndex",
			"--output", inputFile,
			"--input", "a",
			"--documentEquivalenceClassFile", outputFile.toString()
		});
	}
}
