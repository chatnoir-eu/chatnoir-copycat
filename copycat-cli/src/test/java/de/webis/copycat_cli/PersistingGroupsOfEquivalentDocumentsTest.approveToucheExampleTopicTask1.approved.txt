
{"hash":"-746966759","ids":["clueweb12-0008wb-72-04101","clueweb12-0011wb-34-00247","clueweb12-1415wb-38-16623"]}
{"hash":"-2038705227","ids":["clueweb12-0300tw-28-06353","clueweb12-0300tw-28-06354"]}
{"hash":"-1626546300","ids":["clueweb12-0816wb-79-13871","clueweb12-0816wb-91-21490"]}
{"hash":"-806597055","ids":["clueweb12-0909wb-44-10283","clueweb12-0910wb-63-03917"]}
{"hash":"518747447","ids":["clueweb12-0611wb-39-28599","clueweb12-0611wb-83-23505"]}
{"hash":"-71133643","ids":["clueweb12-0611wb-47-24157","clueweb12-0611wb-83-23506"]}
{"hash":"-1242824414","ids":["clueweb12-0011wb-07-25762","clueweb12-0703wb-42-26128","clueweb12-1401wb-64-29120"]}
{"hash":"274258107","ids":["clueweb12-0201wb-68-13660","clueweb12-0400wb-19-11055","clueweb12-1212wb-72-17030"]}
{"hash":"-1758910177","ids":["clueweb12-0211wb-77-26849","clueweb12-0212wb-07-16388","clueweb12-0302wb-06-23242"]}
{"hash":"-1643213096","ids":["clueweb12-0012wb-03-09105","clueweb12-0404wb-14-05843"]}
{"hash":"-1443126058","ids":["clueweb12-1908wb-05-14384","clueweb12-1908wb-10-08307"]}
{"hash":"124319230","ids":["clueweb12-0411wb-47-19875","clueweb12-0503wb-77-33173"]}
{"hash":"591112702","ids":["clueweb12-0611wb-59-18968","clueweb12-0611wb-78-12033"]}
{"hash":"1642910416","ids":["clueweb12-0808wb-00-05668","clueweb12-0813wb-51-21138"]}
{"hash":"333545145","ids":["clueweb12-0003wb-04-33867","clueweb12-0500wb-09-07967"]}
{"hash":"51321795","ids":["clueweb12-0502wb-93-29740","clueweb12-1901wb-12-13419"]}
{"hash":"1137637276","ids":["clueweb12-0611wb-47-24156","clueweb12-0611wb-78-12030"]}
{"hash":"-1569728707","ids":["clueweb12-0611wb-35-13946","clueweb12-0611wb-78-12026"]}
{"hash":"-183525220","ids":["clueweb12-0611wb-35-13948","clueweb12-0611wb-78-12029"]}
{"hash":"-2050982950","ids":["clueweb12-0406wb-88-21897","clueweb12-0406wb-92-10306"]}
{"hash":"1862507614","ids":["clueweb12-0611wb-15-24827","clueweb12-0611wb-66-12767"]}